worker/firewaller/manifold.go

```diff
@@ -86,10 +86,21 @@ func (cfg ManifoldConfig) start(context dependency.Context) (worker.Worker, erro
 	if err := context.Get(cfg.EnvironName, &environ); err != nil {
 		return nil, errors.Trace(err)
 	}
+
+	// Check if the env supports global firewalling.  If the
+	// configured mode is instance, we can ignore fwEnv being a
+	// nil value, as it won't be used.
+	fwEnv, fwEnvOK := environ.(environs.Firewaller)
+
 	mode := environ.Config().FirewallMode()
 	if mode == config.FwNone {
 		logger.Infof("stopping firewaller (not required)")
 		return nil, dependency.ErrUninstall
+	} else if mode == config.FwGlobal {
+		if !fwEnvOK {
+			logger.Infof("Firewall global mode set on provider with no support. stopping firewaller")
```

Comment:
IMO this should be an error, but I guess we have cases where we weren't rejecting FwGlobal, and have a dummy implementation? Oh well, I guess this is OK.